lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -140,3 +148,66 @@ defaultRetryPolicy =
     limitRetriesByCumulativeDelay (60 * second) (exponentialBackoff 10000)
   where
     second = 1000*1000
+
+type RestoreHandle block = (NE.NonEmpty block -> BlockHeader -> IO Bool)
+
+class Cursor s where
+    toHeader :: s -> BlockHeader
+
+restore
+    :: Cursor s
```

Comment:
What is the need for the type-class here :thinking: ? What about passing an `IO BlockHeader` as a parameter instead of `IO s` ?